src/java.base/share/classes/jdk/internal/classfile/CodeBuilder.java

```diff
@@ -0,0 +1,1386 @@
+        with(ConvertInstruction.of(fromType, toType));
+        return this;
+    }
+
+    default CodeBuilder stackInstruction(Opcode opcode) {
+        with(StackInstruction.of(opcode));
+        return this;
+    }
+
+    default CodeBuilder operatorInstruction(Opcode opcode) {
+        with(OperatorInstruction.of(opcode));
+        return this;
+    }
+
+    default CodeBuilder constantInstruction(Opcode opcode, ConstantDesc value) {
+        BytecodeHelpers.validateValue(opcode, value);
+        return with(switch (opcode) {
+            case SIPUSH, BIPUSH -> ConstantInstruction.ofArgument(opcode, ((Number)value).intValue());
+            case LDC, LDC_W, LDC2_W -> ConstantInstruction.ofLoad(opcode, BytecodeHelpers.constantEntry(constantPool(), value));
+            default -> ConstantInstruction.ofIntrinsic(opcode);
+        });
+    }
+
+    default CodeBuilder constantInstruction(ConstantDesc value) {
+        //avoid switch expressions here
+        if (value == null || value == ConstantDescs.NULL)
+            return aconst_null();
+        if (value instanceof Integer iVal)
+            return switch (iVal) {
+                case -1 -> iconst_m1();
+                case  0 -> iconst_0();
+                case  1 -> iconst_1();
+                case  2 -> iconst_2();
+                case  3 -> iconst_3();
+                case  4 -> iconst_4();
+                case  5 -> iconst_5();
+                default -> (iVal >= Byte.MIN_VALUE && iVal <= Byte.MAX_VALUE) ? bipush(iVal)
+                         : (iVal >= Short.MIN_VALUE && iVal <= Short.MAX_VALUE) ? sipush(iVal)
+                         : ldc(constantPool().intEntry(iVal));
+            };
+        if (value instanceof Long lVal)
+            return lVal == 0l ? lconst_0()
+                 : lVal == 1l ? lconst_1()
+                 : ldc(constantPool().longEntry(lVal));
+        if (value instanceof Float fVal)
+            return fVal == 0.0f ? fconst_0()
+                 : fVal == 1.0f ? fconst_1()
+                 : fVal == 2.0f ? fconst_2()
+                 : ldc(constantPool().floatEntry(fVal));
+        if (value instanceof Double dVal)
+            return dVal == 0.0d ? dconst_0()
+                 : dVal == 1.0d ? dconst_1()
+                 : ldc(constantPool().doubleEntry(dVal));
+        return ldc(BytecodeHelpers.constantEntry(constantPool(), value));
+    }
+
+    default CodeBuilder monitorInstruction(Opcode opcode) {
+        with(MonitorInstruction.of(opcode));
+        return null;
+    }
+
+    default CodeBuilder nopInstruction() {
+        with(NopInstruction.of());
+        return this;
+    }
+
+
+    default CodeBuilder nop() {
+        return nopInstruction();
+    }
+
+    // Base pseudo-instruction builder methods
+
+    default Label newBoundLabel() {
+        var label = newLabel();
+        labelBinding(label);
+        return label;
+    }
+
+    default CodeBuilder labelBinding(Label label) {
```

Comment:
Maybe just `bind` or `bindLabel` ?

Reply:
It has been discussed (and changed) many times. Please raise this discussion at classfile-api-dev at openjdk.org